From c4f497793c20ac9842bad3fbead8c4a27af4f211 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 19 Jan 2021 20:37:42 +0200
Subject: [PATCH] [SW-20810] lzo: compilation fixup

In order to utilize the LZO sources taken from u-boot, we need to be
able to build them.

This commit fixes the included header and provide needed macros for
unaligned access.

Change-Id: I0a3c641edf21cbb49e400db7a93bae39550a9b9a
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/lzo/lzo1x_decompress.c b/lzo/lzo1x_decompress.c
index 65fef0b..9b7a1eb 100644
--- a/lzo/lzo1x_decompress.c
+++ b/lzo/lzo1x_decompress.c
@@ -11,10 +11,11 @@
  *  Richard Purdie <rpurdie@openedhand.com>
  */
 
-#include <common.h>
-#include <linux/lzo.h>
-#include <asm/byteorder.h>
-#include <asm/unaligned.h>
+#include <preboot.h>
+#include <string.h>
+
+#include "lzo.h"
+#include "unaligned.h"
 #include "lzodefs.h"
 
 #define HAVE_IP(x, ip_end, ip) ((size_t)(ip_end - ip) < (x))
diff --git a/lzo/unaligned.h b/lzo/unaligned.h
new file mode 100644
index 0000000..9c9dff8
--- /dev/null
+++ b/lzo/unaligned.h
@@ -0,0 +1,31 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2020 HabanaLabs Ltd.
+ */
+
+#ifndef _GENERIC_UNALIGNED_H
+#define _GENERIC_UNALIGNED_H
+
+#include <endian.h>
+#include "unaligned/le_byteshift.h"
+#include "unaligned/be_byteshift.h"
+#include "unaligned/generic.h"
+
+#ifndef __force
+#define __force
+#endif
+
+/*
+ * Select endianness
+ */
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+#define get_unaligned	__get_unaligned_le
+#define put_unaligned	__put_unaligned_le
+#elif __BYTE_ORDER == __BIG_ENDIAN
+#define get_unaligned	__get_unaligned_be
+#define put_unaligned	__put_unaligned_be
+#else
+#error invalid endian
+#endif
+
+#endif
